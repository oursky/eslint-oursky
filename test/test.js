const path = require("path");

const { CLIEngine } = require("eslint");

function patchResults(results) {
  const cwd = process.cwd();
  return results.map(result => {
    return {
      ...result,
      filePath: path.relative(cwd, result.filePath),
    };
  });
}

function runTest(filename) {
  test(filename, () => {
    const cli = new CLIEngine();
    const report = cli.executeOnFiles([path.join("src", filename)]);
    const formatter = cli.getFormatter("unix");
    const results = patchResults(report.results);
    // Patch results so that all paths are relative to cwd.
    const actual = formatter(results);
    expect(actual).toMatchSnapshot();
  });
}

runTest("simple.ts");
runTest("no-scrollTo.ts");
