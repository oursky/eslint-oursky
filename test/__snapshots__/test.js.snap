// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react.tsx 1`] = `
"src/react.tsx:3:26: Use VFC and VoidFunctionComponent instead of FC and FunctionComponent. [Error/@oursky/react-vfc]
src/react.tsx:7:26: Use VFC and VoidFunctionComponent instead of FC and FunctionComponent. [Error/@oursky/react-vfc]
src/react.tsx:11:26: Use VFC and VoidFunctionComponent instead of FC and FunctionComponent. [Error/@oursky/react-vfc]
src/react.tsx:15:26: Use VFC and VoidFunctionComponent instead of FC and FunctionComponent. [Error/@oursky/react-vfc]

4 problems"
`;

exports[`simple.ts 1`] = `
"src/simple.ts:1:1: Unexpected var, use let or const instead. [Error/no-var]
src/simple.ts:1:8: Expected a \`const\` assertion instead of a literal type annotation. [Error/@typescript-eslint/prefer-as-const]
src/simple.ts:5:8: Missing return type on function. [Error/@typescript-eslint/explicit-module-boundary-types]
src/simple.ts:6:11: Switch is not exhaustive. Cases not matched: "b" [Error/@typescript-eslint/switch-exhaustiveness-check]
src/simple.ts:12:1: Unexpected var, use let or const instead. [Error/no-var]
src/simple.ts:12:5: 'b' is assigned a value but never used. Allowed unused vars must match /^_/u. [Error/@typescript-eslint/no-unused-vars]
src/simple.ts:12:11: Unexpected number value in conditional. An explicit zero/NaN check is required. [Error/@typescript-eslint/strict-boolean-expressions]
src/simple.ts:14:8: Function 'thisFunctionIsHardToComprehend' has a complexity of 14. Maximum allowed is 10. [Error/complexity]
src/simple.ts:14:17: Refactor this function to reduce its Cognitive Complexity from 19 to the 10 allowed. [Error/sonarjs/cognitive-complexity]
src/simple.ts:55:25: 'unused' is defined but never used. Allowed unused args must match /^_/u. [Error/@typescript-eslint/no-unused-vars]

10 problems"
`;
